packages/pages/src/common/src/template/internal/loader.ts

```diff
@@ -0,0 +1,73 @@
+import { TemplateModule } from "../types.js";
+import {
+  convertTemplateModuleToTemplateModuleInternal,
+  TemplateModuleInternal,
+} from "./types.js";
+import esbuild from "esbuild";
+import { importFromString } from "module-from-string";
+
+const TEMP_DIR = ".temp";
+
+/**
+ * Loads all templates in the project.
+ * @param templateModulePaths the templates filepaths to load as modules
+ * @param transpile set to true if the templates need to be transpiled (such as when they are in tsx format)
+ * @param adjustForFingerprintedAsset removes the fingerprint portion (for server bundles)
+ * @returns Promise<{@link TemplateModuleCollection}>
+ */
+export const loadTemplateModules = async (
+  templateModulePaths: string[],
+  transpile: boolean,
+  adjustForFingerprintedAsset: boolean
+): Promise<TemplateModuleCollection> => {
+  const importedModules = [] as TemplateModuleInternal<any, any>[];
+  for (const templateModulePath of templateModulePaths) {
+    let templateModule = {} as TemplateModule<any, any>;
+    try {
+      if (transpile) {
+        const buildResult = await esbuild.build({
+          entryPoints: [templateModulePath],
+          outdir: TEMP_DIR,
+          write: false,
+          format: "esm",
+          bundle: true,
+        });
+
+        templateModule = await importFromString(
+          buildResult.outputFiles[0].text
+        );
+      } else {
+        templateModule = await import(templateModulePath);
+      }
+    } catch (e) {
+      throw new Error(`Could not import ${templateModulePath} ${e}`);
+    }
+
+    const templateModuleInternal =
+      convertTemplateModuleToTemplateModuleInternal(
+        templateModulePath,
+        templateModule,
+        adjustForFingerprintedAsset
+      );
+
+    importedModules.push({
+      ...templateModuleInternal,
+      path: templateModulePath,
+    });
+  }
+
+  return importedModules.reduce((prev, module) => {
+    if (prev.has(module.config.name)) {
```

Comment:
at this point we've reconciled the config name right? like we've checked if a `name` was set on the config and otherwise use the filename?

Reply:
This ensures there aren't two features using the same name. This actually came from your code originally.

Reply:
Your response doesn't answer my original question. 

Is the config name on the module at this point specifically what's set in `config.name` in a TemplateConfig or is it the reconciled value of: `config.name if set OR filename`

Reply:
If it's just a copy of what already existed then I assume it worked though in which case you can disregard this.

Reply:
`convertTemplateModuleToTemplateModuleInternal` (called above) handles determining the name to use; it prefers the `name` defined in TemplateConfig and if not set uses the filename.